### DISCUSSION TITLE
How to handle rules with refunds?

### DISCUSSION THREAD

**Discussion by Skaronator** [firefly-iii/firefly-iii#5065] [Firefly III] [answered]
Hi,

just playing around for the first time with the rules in combination with the Firefly CSV importer. The main thing that doesn't make sense for me is how I should handle refunds with rules.

Lets say I have the following rule for Zalando:

![image](https://user-images.githubusercontent.com/2886913/132689029-7280219e-10d1-42f3-8d9c-7fc7577a643c.png)

This rule works perfectly fine when using the CSV importer when I spent money since "Source Account" is defined in the CSV Importer and the rule sets the "Destination Account".

But this rule breaks when I recieve money due to a refund or cenceled order.

This issue is that I can only use "Set destination account to" or "Set source account to" but no "Set other account to". (since Source Account was already defined in the CSV importer)

Is there a elegant solution to my issue or should I just use 2 Rules one for withdraw and one for deposit?

**Comment by Skaronator** [comment, marked as answer]
Ah just noticed that the trigger would be also not correct when its a deposit... so 2 sperate rules is probably the way to go

**Reply by JC5** [reply]
I'm afraid so. From the context of the CSV importer it makes sense, but the rule engine has no "context" so it can't tell which one is the "other one".
